tests/core/test_configuration.py

```diff
@@ -375,6 +377,11 @@ def test_getsection(self):
             test_conf.getsection('testsection'),
         )
 
+        self.assertEqual(
+            OrderedDict([('key', 'value')]),
+            test_conf.getsection('new_section'),
+        )
+
```

Comment:
Shall we have one more assert between `test_conf.getsection('non_existing_section')` and an empty OrderedDict?

Reply:
Good idea

Reply:
Returns None it turns out, but yet lets test it.

Reply:
May be a bit too picky:
but I think it's worthwhile adding one more assert for a section *only* existing in `airflow_defaults`'s sections (and not existing in env var either). In that case, we should expect an empty OrderedDict rather than `None`. 

Just to ensure we cover all cases.

Reply:
Sure, not difficult - let's leave this better than we found it!

Reply:
I realize my last question/request may not make sense... Battery seems too low after a long day ;-)

Reply:
I think I got what you mean -- check now?

Reply:
Oh you did.